bolt/lib/Profile/StaleProfileMatching.cpp

```diff
@@ -640,6 +663,9 @@ void assignProfile(BinaryFunction &BF,
       if (Jump->Flow == 0)
         continue;
 
+      // Skip the artificial sink block
```

Comment:
Can instead use `BlockOrder.size() - 1` to avoid the check on each block.

Reply:
I'm not seeing where "BlockOrder.size() - 1" can be used. This check "if (Jump->Target == Func.Sink) continue;" ensures that the assertion within the index operation here "BinaryBasicBlock &SuccBB = *BlockOrder[Jump->Target - 1];" doesn't fail

Reply:
You're right. Didn't get the full context of this loop.